client/allocrunnerv2/taskrunner/task_runner_hooks.go

```diff
@@ -225,7 +290,10 @@ func (tr *TaskRunner) stop() error {
 		}
 	}
 
-	return nil
+	if len(merr.Errors) == 1 {
```

Comment:
Why do you need this special casing instead of using merr.ErrorOrNil on a single value list?

Reply:
Good question. It's mostly ported from old code, and I think it was just to cleanup log output?

Doesn't seem worth the special casing. I'll remove it.